### PR TITLE
fixed condition when can show password input. Fixed logic for pulling…

### LINKED ISSUE #26903: Guest visitor e-mail in checkout kicks off password field visibility
### Preconditions (*)

1. Magento 2.3.4

### Steps to reproduce (*)

1. add items to cart and reach checkout , use a unregistered e-mail in e-mail field and fill address data
2. go back to cart page (or to any other page as long outside from checkout page)
3. then go back to checkout page and see that it presents password fields for authenticating for e-mail that is not registered

### Expected result (*)

1. add items to cart and reach checkout , use a unregistered e-mail in e-mail field and fill address data
2. go back to cart page (or to any other page as long outside from checkout page)
3. then go back to checkout page and see that password fields are not displayed since e-mail is not jet registered 

### Actual result (*)

1. when reaching checkout and adding a unregistered e-mail and then visiting other pages and going back to checkout it shows password fields for unregistered visitors 

2. this is happening cause of https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/view/frontend/web/js/view/form/element/email.js#L196 and was introduced with https://github.com/magento/magento2/commit/0861a3077135f40d9728dc1e6cfec3ab7387f3a3

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
… email value from persistent storage

### Fixed Issues (if relevant)

1. Fixes magento/magento2#26903 : 
Guest visitor e-mail in checkout kicks off password field visibility

### Manual testing scenarios (*)

1. add items to cart and reach checkout , use a unregistered e-mail in e-mail field and fill address data
2. go back to cart page (or to any other page as long outside from checkout page)
3. then go back to checkout page and see that it presents password fields for authenticating for e-mail that is not registered

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by eduard13** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @eduard13, here is your [new Magento instance](https://pr-27691.instances.magento-community.engineering).
Admin access: [https://pr-27691.instances.magento-community.engineering/admin_6399](https://pr-27691.instances.magento-community.engineering/admin_6399)
Login: `2a4ed3bc` Password: `de96e26a9cdc`
Instance will be terminated in up to 3 hours.

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @ProkopovVitaliy, could you please cover this fix by an automated test? I'd recommend a simple [MFTF](https://devdocs.magento.com/mftf/docs/getting-started.html) test. 
Thank you.

**Comment by ProkopovVitaliy** [comment]
@eduard13 added MFTF test )

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest">
+        <annotations>
+            <features value="Checkout"/>
+            <stories value="Visible password field for unregistered e-mail on Checkout"/>
+            <title value="Visibility password field for unregistered e-mail on Checkout process"/>
+            <description value="Guest should not be able to see password field if entered unregistered email"/>
+            <stories value="Guest Checkout"/>
+            <testCaseId value="MC-14695"/>
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml]: I don't think that there is a specific `testCaseId`. So it's better to remove it.

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest">
+        <annotations>
+            <features value="Checkout"/>
+            <stories value="Visible password field for unregistered e-mail on Checkout"/>
+            <title value="Visibility password field for unregistered e-mail on Checkout process"/>
+            <description value="Guest should not be able to see password field if entered unregistered email"/>
+            <stories value="Guest Checkout"/>
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml]: Please remove this one, as you have declared already a bit upper.

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            <testCaseId value="MC-14695"/>
+            <severity value="CRITICAL"/>
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml]: Please change it to MINOR
```suggestion
            <severity value="MINOR"/>
```

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            <group value="checkout"/>
+            <group value="shoppingCart"/>
+            <group value="mtf_migrated"/>
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml]: Your test doesn't migrate an MTF test, so please remove this tag.

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        </annotations>
+        <before>
+            <createData entity="SimpleTwo" stepKey="simpleProduct"/>
+        </before>
+
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml]: Please remove all the empty lines.

File: app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontOpenCheckoutCartPageActionGroup">
+        <annotations>
+            <description>Goes to the Storefront Checkout Cart page.</description>
+        </annotations>
+
+        <amOnPage url="{{CheckoutPage.url}}/cart" stepKey="openCheckoutCartPage"/>
+        <waitForPageLoad stepKey="waitForCheckoutCartPageLoaded"/>
+        <amOnPage url="{{CheckoutPage.url}}" stepKey="openCheckoutPage"/>
+        <waitForPageLoad stepKey="waitForCheckoutPageLoaded"/>
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml]: What's the reason of going to cart and after that to checkout page?
I'd use the following Action Group instead `StorefrontOpenCheckoutPageActionGroup`.

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        <annotations>
+            <features value="Checkout"/>
+            <stories value="Visible password field for unregistered e-mail on Checkout"/>
+            <title value="Visibility password field for unregistered e-mail on Checkout process"/>
+            <description value="Guest should not be able to see password field if entered unregistered email"/>
+            <stories value="Guest Checkout"/>
+            <testCaseId value="MC-14695"/>
+            <severity value="CRITICAL"/>
+            <group value="checkout"/>
+            <group value="shoppingCart"/>
```
Comment (eduard13) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontVisiblePasswordFieldForUnregisteredEmailOnCheckoutTest.xml]: Don't think that this group is necessary here.

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @ProkopovVitaliy, thank you for the test coverage, could you please review the minor comments below?

File: app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml (hunk 6 above)
Comment (ProkopovVitaliy) [reply app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml]: @eduard13 bug reproduce only when customer add items to cart and reach checkout , use a unregistered e-mail in e-mail field and fill address data after that visits any other page as long outside from checkout page and after that come back to checkout.  

So i decided to take this case into consideration. If your opinion is opposite i can remove this case.

Comment (eduard13) [reply app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml]: If I'm not mistaken, the issue comes up after navigating from checkout to any other page, and then you're coming back to checkout. So there is no required step to be on cart page.
Correct me if I'm wrong.

Comment (ProkopovVitaliy) [reply app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml]: @eduard13 Yes you are right. So you recommend replace `StorefrontOpenCheckoutCartPageActionGroup` to `StorefrontOpenCheckoutPageActionGroup` ?

Comment (eduard13) [reply app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml]: Yes.

Comment (ProkopovVitaliy) [reply app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontOpenCheckoutCartPageActionGroup.xml]: @eduard13 done ) applied all recommendations.

**Review by eduard13 — DISMISSED** [review]
Thank you for collaboration. 👍

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-7312](https://jira.corp.magento.com/browse/ENGCOM-7312) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@ProkopovVitaliy thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**
1. Go to Storefront as a guest;
2. Add product to Cart;
3. Go to Shopping Cart;
4. Click on "**Proceed to Checkout**" button;
5. Fill e-mail and all required fields;
6. Go back to Shopping Cart;
7. Click on "**Proceed to Checkout**" button again;

Before: :heavy_multiplication_x: password field for an unregistered visitor is shown

passwordbefore

**After:** :heavy_check_mark: password fields are not displayed since e-mail is not yet registered

passwordafter

**Comment by ihor-sviziev** [comment]
@VladimirZaets could you explain why this PR was moved to on hold?

**Comment by eduard13** [comment]
Hi @VladimirZaets could you please provides any updates on this PR?
Thank you.

**Review by sdzhepa — APPROVED** [review]

**Comment by sdzhepa** [comment]
@ihor-sviziev @eduard13 

Sorry for the long delay and silence. 
We had several meetings about the initial issue and this PR with PMs developers and UXs.
Long story short, was discussed multiple variants of expected results and changes to the checkout flow(can we accept issue/pr or need to reject or update expected result etc).
 As result was created an internal ticket for future improvement.
So, now I do not see any blocker for delivery this PR

**Pull request merged**